paddle/operators/recurrent_op.h

```diff
@@ -167,7 +167,7 @@ class RecurrentGradientAlgorithm {
 
 class RecurrentOp final : public OperatorBase {
 public:
-  void Init() override;
```

Comment:
The original `Init()` is designed to be called every mini-batch. `RecurrentOp`'s inputs need a complex preprocess, such as get `MultiInput`s from `inputs`.

It works fine after change `Init()` to constructors, because all our unittests just run one-minibatch. 

So, it is better to move `Init()`'s code to `InferShape` to support  preprocess of more than one mini-batch.